BaseImage.py

```diff
@@ -1,134 +1,134 @@
-                        self["warnings"].append(
-                            f"Desired Image Level {dim+1} does not exist! Instead using level {osh.level_count-1}! Downstream output may not be correct")
-                    logging.info(
-                        f"{self['filename']} - \t\tloading image from level {dim} of size {osh.level_dimensions[dim]}")
-                    img = osh.read_region((0, 0), dim, osh.level_dimensions[dim])
-                    self[key] = np.asarray(img)[:, :, 0:3]
-                else:  # assume its an explicit size, *WARNING* this will likely cause different images to have different
-                    # perceived magnifications!
-                    logging.info(f"{self['filename']} - \t\tcreating image thumb of size {str(dim)}")
-                    self[key] = np.array(osh.get_thumbnail((dim, dim)))
-            elif "X" in dim.upper():  # specifies a desired operating magnification
-
-                base_mag = self["base_mag"]
-                if base_mag != "NA":  # if base magnification is not known, it is set to NA by basic module
-                    base_mag = float(base_mag)
-                else:  # without knowing base mag, can't use this scaling, push error and exit
-                    logging.error(
-                        f"{self['filename']}: Has unknown or uncalculated base magnification, cannot specify magnification scale: {base_mag}! Did you try getMag?")
-                    return -1
-
-                target_mag = float(dim.upper().split("X")[0])
-
-                down_factor = base_mag / target_mag
-                level = osh.get_best_level_for_downsample(down_factor)
-                relative_down = down_factor / osh.level_downsamples[level]
-                win_size = 2048
-                win_size_down = int(win_size * 1 / relative_down)
-                dim_base = osh.level_dimensions[0]
-                output = []
-                for x in range(0, dim_base[0], round(win_size * osh.level_downsamples[level])):
-                    row_piece = []
-                    for y in range(0, dim_base[1], round(win_size * osh.level_downsamples[level])):
-                        aa = osh.read_region((x, y), level, (win_size, win_size))
-                        bb = aa.resize((win_size_down, win_size_down))
-                        row_piece.append(bb)
-                    row_piece = np.concatenate(row_piece, axis=0)[:, :, 0:3]
-                    output.append(row_piece)
-
-                output = np.concatenate(output, axis=1)
-                output = output[0:round(dim_base[1] * 1 / down_factor), 0:round(dim_base[0] * 1 / down_factor), :]
-                self[key] = output
-            else:
-                logging.error(
-                    f"{self['filename']}: Unknown image level setting: {dim}!")
-                return -1
-        return self[key]
+import logging

```

Comment:
try to not submit white space changes, makes it difficult to see the true changes. there is likely an option in the UI that will allow this, but for this commit, this may help:

https://stackoverflow.com/questions/3515597/add-only-non-whitespace-changes